### ISSUE TITLE
[chapters.lua] Ability to generate XML chapters and embed them in a MKV file with mkvpropedit

### ISSUE THREAD

**Issue by KonoVitoDa** [mar04/chapters_for_mpv#12]
And I'll also use the opportunity to report a problem I found: the bake-chapters function was intended to embed the chapters file into the video file? Because I used it to embed the chapters into a MKV file and while it worked to generate the new file, no chapters were shown on the new file (at least no the new chapters I made, but the old ones that were already present). I checked the MKV on MKVToolNix and it had the metadata file embedded on it, but seems like MKV needs the XML chapters instead, so it would be much appreciated if you could add support for XML chapter files as well!
And what would make the script even better would be if you also added the ability to use mkvpropedit to directly write the chapters into the MKV, and it can make it in just a second! The CLI commands it uses to remove the old and embed the new chapters are respectively:
```mkvpropedit.exe input.mkv --chapters ""```
```mkvpropedit.exe input.mkv --chapters chapters.xml```

**Comment by mihawk90** [comment]
I don't think it's about XML. I'm not sure what ffmpeg does but it doesn't seem to be embedding the chapters from the `.ffmetadata` file properly.
The `ffmpeg` command the script is running is problematic anyway, it actually deletes tracks from the file.

When checking in MKVToolNix it's not even an attachment, it's just a generic "Global tags" element.

Original:
![image](https://github.com/user-attachments/assets/fe1d67a9-5eba-48fb-81fc-5bac548eb679)

```
 ffprobe <file>.mkv
…
  Chapters:
    Chapter #0:0: start 0.000000, end 406.520000
      Metadata:
        title           : Chapter 1
    Chapter #0:1: start 406.520000, end 948.840000
      Metadata:
        title           : Chapter 2
    Chapter #0:2: start 948.840000, end 1180.880000
      Metadata:
        title           : Chapter 3
    Chapter #0:3: start 1180.880000, end 1204.200000
      Metadata:
        title           : Chapter 4
    Chapter #0:4: start 1204.200000, end 1242.480000
      Metadata:
        title           : Chapter 5
```

After script's "Bake":
![image](https://github.com/user-attachments/assets/eddb6c2a-1986-41be-a2eb-ea2be9f59130)

```
 ffprobe <file>.chapters.mkv
…
  Chapters:
    Chapter #0:0: start 0.000000, end 406.520000
      Metadata:
        title           : Chapter 1
    Chapter #0:1: start 406.520000, end 948.840000
      Metadata:
        title           : Chapter 2
    Chapter #0:2: start 948.840000, end 1180.880000
      Metadata:
        title           : Chapter 3
    Chapter #0:3: start 1180.880000, end 1204.200000
      Metadata:
        title           : Chapter 4
    Chapter #0:4: start 1204.200000, end 1242.480000
      Metadata:
        title           : Chapter 5
```

Note the missing English audio track and PGS subtitles and no change in chapters.

> The CLI commands it uses to remove the old and embed the new chapters are respectively:
> `mkvpropedit.exe input.mkv --chapters ""`
> `mkvpropedit.exe input.mkv --chapters chapters.xml`

You don't actually need to clear the chapters and as noted above, it doesn't have to be an XML either - the `.ffmetadata` file works just fine:
```
 cat <file>.mkv.ffmetadata
;FFMETADATA1
;file=/home/tarulia/Videos/<file>.mkv
[CHAPTER]
START=0
END=84440000000
title=Chapter 1
[CHAPTER]
START=84440000000
END=113240000000
title=Intro
[CHAPTER]
START=113240000000
END=406520000000
title=Episode
[CHAPTER]
START=406520000000
END=948840000000
title=Chapter 2
[CHAPTER]
START=948840000000
END=1180880000000
title=Chapter 3
[CHAPTER]
START=1180880000000
END=1204200000000
title=Chapter 4
[CHAPTER]
START=1204200000000
END=1242480000000
title=Chapter 5%
 mkvpropedit <file>.chapters.mkv --chapters <file>.mkv.ffmetadata 
The file is being analyzed.
The changes are written to the file.
Done.
 ffprobe <file>.chapters.mkv
…
  Chapters:
    Chapter #0:0: start 0.000000, end 84.440000
      Metadata:
        title           : Chapter 1
    Chapter #0:1: start 84.440000, end 113.240000
      Metadata:
        title           : Intro
    Chapter #0:2: start 113.240000, end 406.520000
      Metadata:
        title           : Episode
    Chapter #0:3: start 406.520000, end 948.840000
      Metadata:
        title           : Chapter 2
    Chapter #0:4: start 948.840000, end 1180.880000
      Metadata:
        title           : Chapter 3
    Chapter #0:5: start 1180.880000, end 1204.200000
      Metadata:
        title           : Chapter 4
    Chapter #0:6: start 1204.200000, end 1242.480000
      Metadata:
        title           : Chapter 5
```

As you can see it, it just overwrites the existing chapters (there's probably a way to keep the original as a second "edition" but I didn't bother looking into it because Edition support is questionable at best).

**Comment by mar04** [comment]
> The `ffmpeg` command the script is running is problematic anyway, it actually deletes tracks from the file.

Should be fixed by ca71b2c, thanks for noticing.

**Comment by mar04** [comment]
> I checked the MKV on MKVToolNix and it had the metadata file embedded on it, but seems like MKV needs the XML chapters instead, so it would be much appreciated if you could add support for XML chapter files as well!

You can export to xml now.
